packit_service/worker/checker/copr.py

```diff
@@ -138,3 +142,51 @@ def _pre_check(self) -> bool:
                 )
                 return False
         return True
+
+class AreFilesChanged(Checker, GetCoprBuildJobHelperForIdMixin):
+    """
+    Check if any files under the current package's `paths` field is changed.
+    If not, then just skip the current copr build job.
+    """
+    def get_previous_git_ref(self) -> Optional[str]:
+        """
+        Get the previous git ref if any.
+        """
+        if self.job_config.trigger == JobConfigTriggerType.pull_request:
+            # For PRs return the target branch
+            return self.copr_build_helper.pull_request_object.target_branch
+        elif self.job_config.trigger == JobConfigTriggerType.commit:
+            # For branch commits return the effective `HEAD^` commit
+            current_commit = self.copr_build_helper.metadata.commit_sha
+            return f"{current_commit}^"
+        elif self.job_config.trigger == JobConfigTriggerType.release:
+            # For releases we don't do any checks
+            return None
+
+    def get_files_changed(self, previous_git_ref: str) -> list[Path]:
+        """
+        Get the list of files changed between the current git ref and the previous one.
+        """
+        # FIXME: Implement the relevant git diff
+        return []
```

Comment:
Currently stuck on this part. Isn't the git repo cloned somewhere when the `package_config` was evaluated? I was hoping on using that to do the `git diff`, but I haven't navigated to such a part. Should that be an interface in `ogr` instead?

Reply:
`LocalProjectMixin` should give you `self.local_project.git_repo` which is `git.Repo`.
https://github.com/packit/packit-service/blob/11d4722765650d350cab89ba8ec80fcfd0736cfa/packit_service/worker/mixin.py#L212-L246

Reply:
on our arch meeting we were discussing we could get this rather [via API](https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#get-a-commit), this would mean implementing the method for getting a commit in [ogr](https://github.com/packit/ogr)